### PR TITLE
log: added logging package

### PR DESCRIPTION
**Pull request by oxisto**
This PR introduces a small logging subsystem and a server-side logging interceptor to standardize structured, request-scoped logs across the codebase. The goal is to make RPC-level logging predictable and machine-friendly while providing convenient helpers for attaching contextual attributes to every log.

## Key additions

- `core/log` — a lightweight package providing:
	- `WithAttrs(ctx, ...)` to attach request-scoped attributes to contexts so they are included in subsequent logs.
	- `FindAttr` to locate `slog.Attr` values in slices (returns `(*slog.Attr, bool)`).
	- a small `slog.Handler` and level helpers used in tests and local logger setup.
	- unit tests that follow repository testing conventions.

- `core/server/logging_interceptor.go` — `LoggingInterceptor.logRPCRequest` emits structured logs for RPC handlers with attributes such as:
	- `status` (OK / non-OK), `duration` (processing time), and `payload` (when appropriate).
	- pagination metadata for list operations: `page_size`, `page_token`, `results`, `next_page_token`.
	- request-scoped IDs and attrs pulled from context (via `WithAttrs`).

## Logging style and rationale

- Structured logging via Go's `log/slog` so logs are typed and easy to parse.
- Prefer consistent, named attribute keys so downstream tools (metrics, tracing, log processors) can rely on them.
- Attach common request attributes via `WithAttrs` rather than repeating them in each log call.
- `FindAttr` returns a pointer to avoid copying `slog.Value` and to let tests and handlers inspect group values directly.

## Testing and validation

- Unit tests for the interceptor capture `slog` output with a small test handler and assert expected attributes. Tests reuse existing `orchestratortest` mocks for stable inputs.
- `core/log/context_test.go` includes table-driven tests for `WithAttrs` and `FindAttr` using `core/util/assert`.

### CODE REVIEW

File: core/api/orchestrator/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package orchestrator
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// PayloadRequest describes any kind of requests that carries a certain payload.
+// This is for example a Create/Update request carrying an embedded message,
+// which should be created or updated.
+type PayloadRequest interface {
```
Comment (anatheka) [inline core/api/orchestrator/interfaces.go]: Duplicated file? Already existing in the api folder.

File: core/api/orchestrator/request_extensions.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+//
+// This file is part of Confirmate Core.
+
+package orchestrator
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// GetPayload returns the embedded catalog from [CreateCatalogRequest].
```
Comment (anatheka) [inline core/api/orchestrator/request_extensions.go]: Add all requests.

File: core/api/paginate.go [hunk 3]
```diff
@@ -36,9 +37,36 @@ type PaginatedRequest interface {
...
+// for a List gRPC call.
 type PaginatedResponse interface {
 	GetNextPageToken() string
+	proto.Message
+}
+
+// GetResultsCount returns the count of results from a paginated response.
+// By convention, the first field in a paginated response contains the results slice.
+func GetResultsCount(res PaginatedResponse) int {
+	if res == nil {
```
Comment (anatheka) [inline core/api/paginate.go]: Missing test for `res==nil`.

**Review by anatheka — COMMENTED** [review]
Only some minor things: 
- Test files for the functions/methods Encode(), DecodePageToken() and ListAllPaginated() are missing in api/paginate.go
- Tests for files in log folder missing
- I was thinking about possibly adding a `WithLogLevel` option to configure logging settings. This way, users wouldn't have to deal with the entire configuration right away.
- Would be nice if the default catalogs folder path is highlighted in the log file (e.g., ``). Maybe add the demo_catalog.json from Clouditor or better include the catalogs folder from the security-metrics repo? 
image

File: core/api/orchestrator/orchestrator.proto [hunk 4]
```diff
@@ -590,15 +590,20 @@ message SubscribeRequest {
...
-  CHANGE_TYPE_UNSPECIFIED = 0;
-  // ChangeTypeCreated indicates that a new entity was created/stored in the orchestrator.
-  CHANGE_TYPE_CREATED = 1;
-  // ChangeTypeUpdated indicates that an existing entity was updated/modified in the orchestrator.
-  CHANGE_TYPE_UPDATED = 2;
-  // ChangeTypeDeleted indicates that an existing entity was deleted/removed from the orchestrator.
-  CHANGE_TYPE_DELETED = 3;
+// RequestType is a high-level classification of what happened to an entity. It represents the type
+// of operation performed on the entity.
+enum RequestType {
```
Comment (lebogg) [inline core/api/orchestrator/orchestrator.proto]: I found `ChangeType` more descriptive and accurate than `RequestType`

File: core/api/interfaces.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	proto.Message
+}
+
+// HasId interface for messages that have an ID field.
+type HasId interface {
+	GetId() string
+}
+
+// HasTargetOfEvaluationId interface for messages that have a target_of_evaluation_id field.
+type HasTargetOfEvaluationId interface {
```
Comment (lebogg) [inline core/api/interfaces.go]: Use all caps for acronyms: `ID`

File: core/api/interfaces.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	GetTargetOfEvaluationId() string
+}
+
+// HasToolId interface for messages that have a tool_id field.
+type HasToolId interface {
```
Comment (lebogg) [inline core/api/interfaces.go]: Use all caps for acronyms: `ID`

File: core/server/logging_interceptor.go [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+	// Pagination keys
+	keyPageSize      = "page_size"
+	keyPageToken     = "page_token"
+	keyResults       = "results"
+	keyNextPageToken = "next_page_token"
+
+	// Entity keys
+	keyId                   = "id"
+	keyTargetOfEvaluationId = "target_of_evaluation_id"
+	keyToolId               = "tool_id"
```
Comment (lebogg) [inline core/server/logging_interceptor.go]: ID (all caps)

**Review by lebogg — APPROVED** [review]

File: core/log/levels.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	// Delegate to slog.Level's UnmarshalText for standard levels
+	var slogLevel slog.Level
+	if err := slogLevel.UnmarshalText(data); err != nil {
+		return err
+	}
+	*l = Level(slogLevel)
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (oxisto) [reply core/log/levels.go]: its tested by the json unmarshall test

File: core/persistence/logger.go [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+	"fmt"
+	"log/slog"
+	"time"
+
+	"confirmate.io/core/log"
+	"gorm.io/gorm/logger"
+)
+
+// slogGormLogger integrates GORM's logger with slog.
+// SQL queries are only logged at DEBUG level to reduce noise in production.
```
[the remark this answers is not in the document]
Comment (oxisto) [reply core/persistence/logger.go]: Yes
